config_builder.py

```diff
@@ -0,0 +1,251 @@
+#!/opt/datadog-agent/embedded/bin/python
+'''
+This script is used to generate the configuration of the datadog agent, its
+integrations and other moving parts.
+'''
+
+from os import getenv, environ
+from os.path import isfile
+import logging
+from urllib2 import urlopen, URLError, HTTPError
+from socket import getdefaulttimeout, setdefaulttimeout
+from ConfigParser import ConfigParser
+import yaml
+
+class ConfBuilder(object):
+    '''
+    This class manages the configuration files
+    '''
+    def __init__(self):
+        # excludes from the generic variables parsing the ones that have a
+        # certain logic warpped around them
+        self.exclude_from_generic = [
+            'DD_API_KEY', 'DD_API_KEY_FILE', 'DD_HOME',
+            'DD_START_AGENT', 'DD_LOGS_STDOUT'
+            ]
+        dd_agent_root = '/etc/dd-agent'
+        dd_home = getenv('DD_HOME')
+        if dd_home is not None:
+            dd_agent_root = '{}/agent'.format(dd_home)
+        self.datadog_conf_file = '{}/datadog.conf'.format(dd_agent_root)
+        self.supervisor_conf_file = '{}/supervisor.conf'.format(dd_agent_root)
+        self.conf_d = '{}/conf.d'.format(dd_agent_root)
+        # This will store the config parser object that is used in the different functions
+        self.config = None
+
+    def set_instance_property(self, integration, property_key, property_value):
```

Comment:
> Not sure what's hacky in that

Fair enough, I wasn't clear here. It certainly is more flexible, but this is bound to fail in unexpected ways as config files change, and we observed that this kind of config-level failures tend to confuse users a lot, leading to frustration, and unnecessary support tickets.
It will also greatly limit our freedom in revamping config files in fear that someone is editing them with an ad-hoc system like this one.

> It looks more complex but in the long term it would be more maintainable than having to change 6 sed in 6 entrypoint.sh

Again, we don't expect this to grow much (but rather to shrink with agent 6) and we'll reduce the number of entrypoint files. Integration config should be modified by passing whole files, not individual settings. Those we have in the entrypoint now are special cases that we want to get rid of, not generalize.